services/traction/api/protocols/v1/connection/connection_protocol.py

```diff
@@ -42,34 +42,33 @@ async def notify(self, profile: Profile, event: Event):
-        self.logger.debug(f"payload={payload}")
 
         await self.before_all(profile=profile, payload=payload)
 
         if role_match(self.role, their_role):
 
             await self.before_any(profile=profile, payload=payload)
-
-            if ConnectionStateType.start == payload["state"]:
-                await self.on_start(profile=profile, payload=payload)
-            elif ConnectionStateType.init == payload["state"]:
-                await self.on_init(profile=profile, payload=payload)
-            elif ConnectionStateType.invitation == payload["state"]:
-                await self.on_invitation(profile=profile, payload=payload)
-            elif ConnectionStateType.request == payload["state"]:
-                await self.on_request(profile=profile, payload=payload)
-            elif ConnectionStateType.response == payload["state"]:
-                await self.on_response(profile=profile, payload=payload)
-            elif ConnectionStateType.active == payload["state"]:
-                await self.on_active(profile=profile, payload=payload)
-            elif ConnectionStateType.completed == payload["state"]:
-                await self.on_completed(profile=profile, payload=payload)
-            elif ConnectionStateType.abandoned == payload["state"]:
-                await self.on_abandoned(profile=profile, payload=payload)
-            elif ConnectionStateType.error == payload["state"]:
-                await self.on_error(profile=profile, payload=payload)
-            else:
-                pass
+            if await self.approve_for_processing(profile=profile, payload=payload):
+                if ConnectionStateType.start == payload["state"]:
+                    await self.on_start(profile=profile, payload=payload)
+                elif ConnectionStateType.init == payload["state"]:
+                    await self.on_init(profile=profile, payload=payload)
+                elif ConnectionStateType.invitation == payload["state"]:
+                    await self.on_invitation(profile=profile, payload=payload)
+                elif ConnectionStateType.request == payload["state"]:
+                    await self.on_request(profile=profile, payload=payload)
+                elif ConnectionStateType.response == payload["state"]:
+                    await self.on_response(profile=profile, payload=payload)
+                elif ConnectionStateType.active == payload["state"]:
+                    await self.on_active(profile=profile, payload=payload)
+                elif ConnectionStateType.completed == payload["state"]:
+                    await self.on_completed(profile=profile, payload=payload)
+                elif ConnectionStateType.abandoned == payload["state"]:
+                    await self.on_abandoned(profile=profile, payload=payload)
+                elif ConnectionStateType.error == payload["state"]:
+                    await self.on_error(profile=profile, payload=payload)
+                else:
```

Comment:
This will apply to most of your comments... yes all this needs to be refactored. we have "completed" the API and now we need to reduce code and optimize it for functionality and maintainability. one huge roadblock has been deferring the message queue, so i'd say we consider that before refactoring - or at least refactor in such a way that we can slide it in fairly easily.